src/v/kafka/server/handlers/delete_topics.cc

```diff
@@ -81,23 +83,57 @@ delete_topics_handler::handle(request_context ctx, ss::smp_service_group) {
 
     std::vector<cluster::topic_result> res;
 
+    // Measure the partition mutation rate
+    auto resp_delay = 0ms;
+    auto quota_exceeded_it = co_await ssx::partition(
+      request.data.topic_names.begin(),
+      request.data.topic_names.end(),
+      [&ctx, &resp_delay](const model::topic& t) -> ss::future<bool> {
+          const auto cfg = ctx.metadata_cache().get_topic_cfg(
+            model::topic_namespace_view(model::kafka_namespace, t));
+          const auto mutations = cfg ? cfg->partition_count : 0;
+          const auto delay
+            = co_await ctx.quota_mgr().record_partition_mutations(
+              ctx.header().client_id, mutations);
+          resp_delay = std::max(delay, resp_delay);
```

Comment:
unsafe to access resp_delay?

Reply:
same here too, `delete_topics_request::handler` is a coroutine

Reply:
fixed